### PR TITLE
Updating remaining files to Typescript

### PR DESCRIPTION
**Pull request by devinmatte**
**95% Typescript!**

We're now almost entirely typescript!

Tested locally, everything is happy

After this PR I will consider the Typescript migration done

### CODE REVIEW

**Review by Smilerk — APPROVED** [review]
🔥

File: src/prerender.ts [hunk 1]
```diff
@@ -61,12 +62,12 @@ const prerenderRoute = (shape, stationIds) => {
     };
 
     entries.forEach((entry) => {
-        if (entry.type === 'stationRange') {
+        if (typeof entry !== 'function' && entry.type === 'stationRange') {
```
Comment (mathcolo) [inline src/prerender.ts:65]: Just curious why this type check?

File: package.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
     "name": "new-train-tracker",
     "version": "2.2.0",
     "description": "New Train Tracker by TransitMattters",
-    "main": "index.js",
+    "main": "index.tsx",
```
Comment (mathcolo) [inline package.json:5]: Just curious, does `main` trip up on a TS entrypoint?

File: src/components/Line.tsx [hunk 3]
```diff
@@ -48,14 +56,13 @@ const getRouteColor = (colors, routeId, focusedRouteId) => {
...
 
-export const Line = (props) => {
-    const { api, line } = props;
+export const Line: React.FC<LineProps> = ({ api, line }) => {
     const { getStationLabelPosition, fixedTrainLabelPosition } = line;
     const { stationsByRoute, trainsByRoute, routesInfo } = api;
-    const [viewbox, setViewbox] = useState(undefined);
-    const [svg, setSvg] = useState(null);
+    const [viewbox, setViewbox] = useState<string>();
+    const [svg, setSvg] = useState<SVGSVGElement | null>(null);
```
Comment (mathcolo) [inline src/components/Line.tsx:63]: `SVGSVGElement` lol

**Review by mathcolo — APPROVED** [review]

File: src/prerender.ts (hunk 1 above)
Comment (devinmatte) [reply src/prerender.ts:65]: So there are two things passed in, either an object or a function. And the functions don't have a `type`, so they were making TS angry. The resulting check here is the same, but we now fail quicker if it's a function and Typescript is happy

**Pull request merged**